src/tilesource.js

```diff
@@ -666,6 +670,27 @@ $.TileSource.prototype = {
         return {};
     },
 
+    /**
+     * A tile can have either context2D defined (TileSource.prototype.getContext2D)
```

Comment:
I still think this is confusing. I think it's better to explain what this is for first and then explain the exceptions where it's not used, rather than leading with the exceptions.